contracts/Forwarder.sol

```diff
@@ -36,40 +38,114 @@ contract Forwarder is AccessControlEnumerable, Pausable {
         resultManager = _resultManager;
     }
 
-    /// @notice Set collection payload
-    /// @dev Allows admin to set collection payload
-    /// @param _collectionName keccak256 hash of collection name
-    /// @param _payload payload to call
-    function setCollectionPayload(
-        bytes32 _collectionName,
-        bytes memory _payload
+    /// @notice Set resultGetter Selector
+    /// @dev Allows admin to set resultGetter Selector
+    /// @param _resultGetterSelector resultGetter Selector
+    function setResultGetterSelector(
+        bytes4 _resultGetterSelector
     ) external onlyRole(FORWARDER_ADMIN_ROLE) {
-        collectionPayload[_collectionName] = _payload;
+        resultGetterSelector = _resultGetterSelector;
     }
 
+    /// @notice Set update selector
+    /// @dev Allows admin to set update selector
+    /// @param _updateSelector update selector
+    function setUpdateSelector(bytes4 _updateSelector)
+        external
+        onlyRole(FORWARDER_ADMIN_ROLE)
+    {
+        updateSelector = _updateSelector;
+    }
+
+    /// @notice Set validate selector
+    /// @dev Allows admin to set validate selector
+    /// @param _validateSelector validate selector
+    function setValidateSelector(bytes4 _validateSelector)
+        external
+        onlyRole(FORWARDER_ADMIN_ROLE)
+    {
+        validateSelector = _validateSelector;
+    }
+
+    /// @notice pause the contract
     function pause() external onlyRole(PAUSE_ROLE) {
         Pausable._pause();
     }
 
+    /// @notice unpause the contract
     function unpause() external onlyRole(PAUSE_ROLE) {
         Pausable._unpause();
     }
 
-    /// @notice get result by collection name
-    function getResult(bytes32 collectionName)
+    /**
+     * @notice Updates the result based on the provided data and returns the latest result
+     * @param data bytes data required to update the result
+     * @return result of the collection, its power and timestamp
+     */
+    function getResult(bytes calldata data)
```

Comment:
should we rename this function to `getLatestResult` or `updateAndGetResult`, so that its implicit that this fn call would update and return the latest result always. 

As opposed to `getResult(bytes32 name)` which would return the result even if its not updated recently or stale. (this way might cost less gas, but stale results possible) @SamAg19 @SkandaBhat

Reply:
Renamed to `getLatestResult` for now